File: exercise/day10/src/main/java/games/FizzBuzz.java
package games;

public class FizzBuzz {
    public static final int MIN = 0;
    public static final int MAX = 100;
    public static final int FIZZ = 3;
    public static final int BUZZ = 5;
    public static final int FIZZBUZZ = 15;

    private FizzBuzz() {
    }

    public static String convert(Integer input) throws OutOfRangeException {
        if (isOutOfRange(input)) {
            throw new OutOfRangeException();
        }
        return convertSafely(input);
    }

    private static String convertSafely(Integer input) {
        if (is(FIZZBUZZ, input)) {
            return "FizzBuzz";
        }
        if (is(FIZZ, input)) {
            return "Fizz";
        }
        if (is(BUZZ, input)) {
            return "Buzz";
        }
        return input.toString();
    }

    private static boolean is(Integer divisor, Integer input) {
        return input % divisor == 0;
    }

    private static boolean isOutOfRange(Integer input) {
        return input <= MIN || input > MAX;
    }
}

File: solution/c#/Day02/Day02/FizzBuzz.cs
namespace Day02
{
    public static class FizzBuzz
    {
        private const int Min = 0;
        private const int Max = 100;
        private const int Fizz = 3;
        private const int Buzz = 5;
        private const int Fizz_Buzz = 15;

        public static string Convert(int input)
            => IsOutOfRange(input)
                ? throw new OutOfRangeException()
                : ConvertSafely(input);

        private static string ConvertSafely(int input)
        {
            if (Is(Fizz_Buzz, input))
                return "FizzBuzz";
            if (Is(Fizz, input))
                return "Fizz";
            if (Is(Buzz, input))
                return "Buzz";

            return input.ToString();
        }

        private static bool Is(int divisor, int input) => input % divisor == 0;

        private static bool IsOutOfRange(int input) => input is <= Min or > Max;
    }
}